SI/Sem.05/Pract.05/Task02.cpp

```diff
@@ -0,0 +1,64 @@
+﻿#include <iostream>
+using namespace std;
+
+
+/*
+ *Напишете програма, която подрежда елементите на редица от N числа, като първите елементи са четните, вторите - нечетните.
+Бонус: Сортирайте четните във възходящ ред,а нечетните в низходящ ред(отново четните са първи,а нечетните - втори).
+
+ */
+
+void sortArray(int (& arr)[], const int SIZE)
+{
+	int currArr[SIZE];
+	int counter = 0;
+
+	for (int i = 0; i < SIZE; ++i)
+	{
+		if (arr[i] % 2 == 0)
+		{
+			currArr[counter++] = arr[i];
+		}
+
+	}
+
+	for (int i = 0; i < SIZE; ++i)
+	{
+		if (arr[i] % 2 != 0)
+		{
+			currArr[counter++] = arr[i];
+		}
+	}
+
+	for (int i = 0; i < SIZE; ++i)
+	{
+		int currEl = currArr[i];
+
+		for (int j = counter + 1; j < SIZE; ++j)
+		{
+			if (currArr[j] < currEl && currArr[j] % 2 == 0)
+			{
+				currArr[counter] = currArr[j];
+				currArr[j] = currEl;
+			}
+
+			else if (currArr[j] % 2 != 0 && currArr[j] > currEl)
+			{
+				currArr[counter] = currArr[j];
+				currArr[j] = currEl;
+			}
+		}
```

Comment:
You should have a sort function, which accepts array, startIndex and endIndex, which sorts the arrays between the two indexes